include/boost/geometry/strategies/agnostic/hull_graham_andrew.hpp

```diff
@@ -116,48 +100,35 @@ struct get_extremes
                 left = *left_it;
             }
 
-            if (greater(*right_it, right))
+            if (less(right, *right_it))
             {
                 right = *right_it;
             }
         }
-    }
-};
+    });
+}
 
 
 template
 <
-    typename InputRange,
-    typename RangeIterator,
+    typename Geometry,
+    typename Point,
     typename Container,
     typename SideStrategy
 >
-struct assign_range
+inline void assign_ranges(Geometry const& geometry,
+                          Point const& most_left, Point const& most_right,
+                          Container& lower_points, Container& upper_points,
+                          SideStrategy const& side)
 {
-    Container lower_points, upper_points;
-
-    typedef typename point_type<InputRange>::type point_type;
-
-    point_type const& most_left;
-    point_type const& most_right;
-
-    inline assign_range(point_type const& left, point_type const& right)
-        : most_left(left)
-        , most_right(right)
-    {}
-
-    inline void apply(InputRange const& range)
+    geometry::detail::for_each_range(geometry, [&](auto const& range)
     {
-        typedef SideStrategy side;
-
         // Put points in one of the two output sequences
-        for (RangeIterator it = boost::begin(range);
-            it != boost::end(range);
-            ++it)
+        for (auto it = boost::begin(range); it != boost::end(range); ++it)
```

Comment:
Also here a range based loop instead?

Reply:
Same here, we require `Range` adapted to Boost.Range concept which is not required to work with range-based for loops.